heartbeat/LVM

```diff
@@ -311,18 +311,15 @@ LVM_status() {
 		fi
 	fi
 	
-	if [ -d /dev/$1 ]; then
-		test "`cd /dev/$1 && ls`" != ""
-		rc=$?
-		if [ $rc -ne 0 ]; then
-			ocf_exit_reason "VG $1 with no logical volumes is not supported by this RA!"
-		fi
-	fi
-
-	if [ $rc -ne 0 ]; then
+	# Ask lvm whether the volume group is active.  This maps to
+	# the question "Are there any logical volumes that are active in
+	# the specified volume group?".
+	lvs --noheadings -o selected  -S lv_active=active,vg_name=${1}|grep -q 1
```

Comment:
Due to timing issues on various storage configurations, all lvm commands were banned from the status/monitor path. There were reports about the lvm programs taking unusually long time to run and that could unfortunately also be due to another VG.